expression/schema.go

```diff
@@ -210,11 +211,13 @@ func (s *Schema) ColumnsByIndices(offsets []int) []*Column {
 
 // GetTypes creates a field type slice.
 func (s *Schema) GetTypes() []*types.FieldType {
-	fields := make([]*types.FieldType, len(s.Columns))
-	for i := range fields {
-		fields[i] = s.Columns[i].RetType
+	if s.fieldTypes == nil {
```

Comment:
is GetTypes thread-safe?

Reply:
nice catch, it's not thread safe but can be called concurrently.

Reply:
init the buf in the NewSchema is ok.